packages/odata-service-inquirer/src/prompts/connectionValidator.ts

```diff
@@ -90,10 +111,95 @@ export class ConnectionValidator {
      *
      * @returns the current connections service provider
      */
-    public get serviceProvider(): ServiceProvider {
+    public get serviceProvider(): ServiceProvider | undefined {
         return this._serviceProvider;
     }
 
+    /**
+     * The auth type used to create an authenticated connection to the system.
+     *
+     * @returns the system auth type
+     */
+    public get systemAuthType(): SystemAuthType | undefined {
+        return this._systemAuthType;
+    }
+    /**
+     * The auth type used to create an authenticated connection to the system.
+     *
+     * @param value the system auth type
+     */
+    public set systemAuthType(value: SystemAuthType) {
+        this._systemAuthType = value;
+    }
+
+    /**
+     * Get the validated url. This is the url that has been successfully validated (not necessarily connected). Use validity to check if the url is reachable.
+     *
+     * @returns the validated url
+     */
+    public get validatedUrl(): string | undefined {
+        return this._validatedUrl;
+    }
+
+    /**
+     * Get the service info used to connect to the system.
+     *
+     * @returns the service info
+     */
+    public get serviceInfo(): ServiceInfo | undefined {
+        return this._serviceInfo;
+    }
+
+    /**
+     * Set the service info used to connect to the system.
+     *
+     * @param serviceInfo the service info
+     */
+    public set serviceInfo(serviceInfo: ServiceInfo) {
+        this._serviceInfo = serviceInfo;
+    }
+
+    /**
+     * Get the connected user name.
+     *
+     * @returns the connected user name
+     */
+    public get connectedUserName(): string | undefined {
+        return this._connectedUserName;
+    }
+
+    /**
+     * Get the refresh token.
+     *
+     * @returns the refresh token
+     */
+    public get refreshToken(): string | undefined {
+        return this._refreshToken;
+    }
+
+    /**
+     * Get the connected system name. If previously set this will be used, otherwise the name is determined
+     * by the system auth type, or the validated url.
+     *
+     * @returns the connected system name
+     */
+    public get connectedSystemName(): string | undefined {
```

Comment:
Connect system name i.e. selected SAP BTP destination name?

Reply:
Not yet no, this will be the system id, the url or a name provided by the user. Dests are not supported yet.